debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -564,4 +571,49 @@ public String quotedTableIdString(TableId tableId) {
     private String replaceDatabaseNamePlaceholder(String sql, String databaseName) {
         return sql.replace(DATABASE_NAME_PLACEHOLDER, databaseName);
     }
+
+    public void callbackOnReadingNewChangeTable(SqlServerPartition partition, ChangeTable table) throws SQLException {
+        final String query = replaceDatabaseNamePlaceholder(START_READING_CHANGE_TABLE,
+                partition.getDatabaseName());
+        final String startReadingChangeTableStmt = String.format(query, table.getCaptureInstance());
+        LOGGER.trace(
+                "Calling the StartReadingFromCaptureInstance stored procedure for database: {} with capture instance: {}",
+                partition.getDatabaseName(),
+                table.getCaptureInstance());
+        execute(startReadingChangeTableStmt);
+    }
+
+    public void callbackOnSnapshotStarted(SqlServerPartition partition) throws SQLException {
+        final String statement = replaceDatabaseNamePlaceholder(START_SNAPSHOT, partition.getDatabaseName());
+        LOGGER.trace(
+                "Calling the StartSnapshot stored procedure for database: {}",
+                partition.getDatabaseName());
```

Comment:
Since trace logging is unlikely to be enabled, it's better to pass the entire `partition` to the log message and let its `toString()` implementation be called only when necessary. Otherwise, `partition.getDatabaseName()` is always invoked, even when not necessary.

Reply:
From what I understand, slf4j uses lazy string evaluation and this isn't necessary.

Reply:
The expression being passed to the logger is evaluated before being passed to the logger.

Reply:
Actually, the database name will be available in the logging context, so it's not necessary to log it explicitly. Otherwise, to preserve the original message, the call could be wrapped in `if (LOGGER.isTraceEnabled())`.

Reply:
You're right. I updated it to use `partition` instead of `partition.getDatabaseName()`. Things I wish SonarLint would catch ...

Reply:
> I updated it to use `partition` instead of `partition.getDatabaseName()`.

But is it necessary? See https://github.com/sugarcrm/debezium/pull/104#discussion_r788198079.

Reply:
It's not necessary. I'll remove.